src/cli.js

```diff
@@ -353,6 +353,65 @@ if (args['--help']) {
 
 run()
 
+/// ---
+
+/** @type {import('@parcel/css')} */
+let parcelCss
+
+/** @returns {import('@parcel/css')} */
+function loadParcelCss() {
+  if (parcelCss) {
+    return parcelCss
+  }
+
+  // Try to load a local version first
+  try {
+    return parcelCss = require('@parcel/css')
+  } catch {}
+
+  return parcelCss = lazyParcelCss()
+}
+
+/**
+ *
+ * @param {boolean} shouldMinify
+ * @param {import('postcss').Result} result
+ * @returns {import('postcss').Result}
+ */
+async function minifyCss(shouldMinify, result) {
+  if (! shouldMinify) {
+    return result
+  }
+
+  let css = loadParcelCss()
+  let transformed
+
+  try {
+    transformed = css.transform({
+      filename: result.opts.from || 'input.css',
+      code: Buffer.from(result.css, 'utf-8'),
+      minify: true,
+      sourceMap: true,
+      sourceMap: !!result.map,
+      inputSourceMap: result.map ? result.map.toString() : undefined,
+    })
+  } catch (err) {
+    console.error("Unable to minify CSS. Using unminified version instead.")
+    console.error(err)
+
+    return result
+  }
+
+  return Object.assign(result, {
+    css: transformed.code.toString('utf8'),
+    map: result.map ? Object.assign(result.map, {
+      toString() {
+        return transformed.map.toString()
+      }
+    }) : result.map,
+  })
+}
```

Comment:
@syntaxai can we rewrite the above `return Object.assign...` with `return {...result, css: transformed.code.toString('utf8'), map: result.map ? {toString: () => transformed.map.toString()} : result.map}` ? Is it a better solution? Is there performance difference?